iroh/src/commands.rs

```diff
@@ -159,132 +267,85 @@ impl Cli {
-        #[clap(long, default_value_t = ProviderRpcPort::Enabled(DEFAULT_RPC_PORT))]
-        rpc_port: ProviderRpcPort,
-        /// Use a token to authenticate requests for data
-        ///
-        /// Pass "random" to generate a random token, or base32-encoded bytes to use as a token
-        #[clap(long)]
-        request_token: Option<RequestTokenOptions>,
+        command: BlobCommands,
     },
-    /// List availble content on the provider.
-    #[clap(subcommand)]
-    List(self::list::Commands),
-    /// Validate hashes on the running provider.
-    Validate {
-        /// RPC port of the provider
-        #[clap(long, default_value_t = DEFAULT_RPC_PORT)]
-        rpc_port: u16,
-        /// Repair the store by removing invalid data
-        #[clap(long, default_value_t = false)]
-        repair: bool,
+    /// Manage a running Iroh node
+    Node {
+        #[clap(subcommand)]
+        command: NodeCommands,
     },
-    /// Shutdown provider.
+}
+
+#[derive(Subcommand, Debug, Clone)]
+pub enum NodeCommands {
+    /// Get status of the running node.
+    Status,
+    /// Get statistics and metrics from the running node.
+    Stats,
+    /// Shutdown the running node.
     Shutdown {
         /// Shutdown mode.
         ///
         /// Hard shutdown will immediately terminate the process, soft shutdown will wait
         /// for all connections to close.
         #[clap(long, default_value_t = false)]
         force: bool,
-        /// RPC port of the provider
-        #[clap(long, default_value_t = DEFAULT_RPC_PORT)]
-        rpc_port: u16,
-    },
-    /// Identify the running provider.
-    Id {
-        /// RPC port of the provider
-        #[clap(long, default_value_t = DEFAULT_RPC_PORT)]
-        rpc_port: u16,
     },
+}
+
+impl NodeCommands {
+    pub async fn run(self, client: RpcClient) -> Result<()> {
+        match self {
+            Self::Shutdown { force } => {
+                client.rpc(ShutdownRequest { force }).await?;
+            }
+            Self::Stats => {
+                let response = client.rpc(StatsGetRequest {}).await??;
+                for (name, details) in response.stats.iter() {
+                    println!(
+                        "{:23} : {:>6}    ({})",
+                        name, details.value, details.description
+                    );
+                }
+            }
+            Self::Status => {
+                let response = client.rpc(StatusRequest).await?;
+
+                println!("Listening addresses: {:#?}", response.listen_addrs);
+                println!("PeerID: {}", response.peer_id);
+            }
+        }
+        Ok(())
+    }
+}
+
+impl RpcCommands {
+    pub async fn run(self, client: RpcClient, env: ConsoleEnv) -> Result<()> {
+        match self {
+            Self::Node { command } => command.run(client).await,
+            Self::Blob { command } => command.run(client).await,
+            Self::Sync(command) => command.run(client, env).await,
```

Comment:
would be nice to unify this to be also `Sync:: { command }`

Reply:
Done :+1: